guides/source/api_app.md

```diff
@@ -0,0 +1,435 @@
+Using Rails for API-only Apps
+-----------------------------
```

Comment:
Change from `-` to `=`. It is good to base this guide at https://raw.githubusercontent.com/spastorino/rails/rails-api/guides/source/action_view_overview.md